src/Runner.Listener/Runner.cs

```diff
@@ -457,6 +461,35 @@ private async Task<int> RunAsync(RunnerSettings settings, bool runOnce = false)
                                     }
                                 }
                             }
+                            // Broker flow
+                            else if (string.Equals(message.MessageType, JobRequestMessageTypes.RunnerJobRequest, StringComparison.OrdinalIgnoreCase))
+                            {
+                                if (autoUpdateInProgress || runOnceJobReceived)
+                                {
+                                    skipMessageDeletion = true;
+                                    Trace.Info($"Skip message deletion for job request message '{message.MessageId}'.");
+                                }
+                                else
+                                {
+                                    var messageRef = StringUtil.ConvertFromJson<MessageRef>(message.Body);
+
+                                    // Create connection
+                                    var credMgr = HostContext.GetService<ICredentialManager>();
+                                    var creds = credMgr.LoadCredentials();
+
+                                    // todo: add retries
```

Comment:
todo before merge

Reply:
actually, i wonder whether we should just let the other while loop handle it. Thoughts?

Reply:
If you want to use the while loop, you need to catch any exception that you want to retry and skip the message deletion, you need to let the MessageListener not bump its `lastMessageId` since it will always try to pull the latest message that it haven't process before.